hcl/hclsyntax/parser.go

```diff
@@ -1711,21 +1711,21 @@ Token:
 	return ret.String(), hcl.RangeBetween(oQuote.Range, cQuote.Range), diags
 }
 
-// decodeStringLit processes the given token, which must be either a
+// ParseStringLiteralToken processes the given token, which must be either a
 // TokenQuotedLit or a TokenStringLit, returning the string resulting from
 // resolving any escape sequences.
 //
 // If any error diagnostics are returned, the returned string may be incomplete
 // or otherwise invalid.
-func (p *parser) decodeStringLit(tok Token) (string, hcl.Diagnostics) {
+func ParseStringLiteralToken(tok Token) (string, hcl.Diagnostics) {
```

Comment:
I'm not sure... Should we move `ParseStringLiteralToken` to `public.go` ?
or should we leave `decodeStringLit` here, add `ParseStringLiteralToken` to `public.go` and call `decodeStringLit` from `ParseStringLiteralToken` ?

Reply:
I think let's leave `ParseStringLiteralToken` here for now because otherwise we'll make it hard to traverse the git history of this function with tools like `git annotate`.